pkg/nonkube/api/site_state.go

```diff
@@ -247,8 +247,8 @@ func (s *SiteState) linkMap(sslProfileBasePath string) site.LinkMap {
 	return linkMap
 }
 
-func (s *SiteState) bindings() *site.Bindings {
-	b := site.NewBindings()
+func (s *SiteState) bindings(sslProfileBasePath string) *site.Bindings {
+	b := site.NewBindings(path.Join(sslProfileBasePath, "certificates/bindings"))
```

Comment:
If you set its path to `certificates/server` instead of `certificates/bindings`, it should work ok.
Even better if you use `api.CertificatesServerPath` instead of the literal string (from: `pkg/nonkube/api/environment.go`).

Reply:
Actually, it would be better to apply the following patch to site_state.go:
https://gist.github.com/fgiorgetti/9eb26360bf26741e6e9893067d7b80fb

This way it will use client certificates for Connectors and server certificates for Listeners.
WDYT?

Reply:
> Actually, it would be better to apply the following patch to site_state.go: https://gist.github.com/fgiorgetti/9eb26360bf26741e6e9893067d7b80fb
> 
> This way it will use client certificates for Connectors and server certificates for Listeners. WDYT?

I expect that the credentials for connectors and listeners will be supplied by the user.

Reply:
And I believe that your expected approach is correct.

So the patch above ensures that the existing mechanism defined through [PROVIDED_CERTIFICATES.md](https://github.com/skupperproject/skupper/blob/v2/cmd/bootstrap/PROVIDED_CERTIFICATES.md),
which allows users to provide their own certificates, is used properly for the respective Listeners and
Connectors, when the tlsCredentials field is populated.

Or do you have concerns with the current structure for user provided certificates?

Reply:
From that doc

```Server and client certificates can be provided whenever your site definition contains at least one RouterAccess (resource).

The expected directory names for the server and client certificates, is determined based on the values of RouterAccess.spec.tlsCredentials (optional field), or RouterAccess.name (default).
```

I assumed those locations were for the inter-router certificates and that those for bindings would be supplied elsewhere.

The same secret could be used for both a connector and a listener. Having them under the same path makes that simpler.

The patch above seems to create a new bindings object for every connector/listener which I don't think will work. (Having everything under the same path avoids the need for that also).

Reply:
Alright, let's keep it as is (using `bindings/`) and I can make further changes to load properly.
Today we also allow "secrets" to be provided, which is currently just used for `Links`. Maybe we
can also extend it to work with Listeners and Connectors as well, in an upcoming PR.